### PR TITLE
Make `PostgresDecodingError.init` public

### PR DESCRIPTION
**Pull request by fabianfett**

### CODE REVIEW

File: Sources/PostgresNIO/New/PSQLError.swift [hunk 1]
```diff
@@ -432,8 +432,18 @@ public struct PostgresDecodingError: Error, Equatable {
     /// The line the decoding was attempted in
     public let line: Int
 
-    @usableFromInline
-    init(
+    /// Create a new ``PostgresDecodingError``
+    /// - Parameters:
+    ///   - code: The reason for the decoding error
+    ///   - columnName: The cell's column name for which the decoding failed
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PSQLError.swift:438]:
```suggestion
    ///   - columnName: The cell's column name for which the decoding failed. If no column name is available, use "".
```

File: Sources/PostgresNIO/New/PSQLError.swift [hunk 2]
```diff
@@ -432,8 +432,18 @@ public struct PostgresDecodingError: Error, Equatable {
...
+    ///   - columnIndex: The cell's column index for which the decoding failed
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PSQLError.swift:439]:
```suggestion
    ///   - columnIndex: The cell's column index for which the decoding failed. If no index is available or applicable, use -1.
```

File: Sources/PostgresNIO/New/PSQLError.swift [hunk 3]
```diff
@@ -432,8 +432,18 @@ public struct PostgresDecodingError: Error, Equatable {
...
+    ///   - targetType: The Swift type which the data should have been decoded to
+    ///   - postgresType: The ``PostgresDataType`` as sent from the database
+    ///   - postgresFormat: The ``PostgresFormat`` the data is received in
+    ///   - postgresData: The actual bytes that were attempted to decode
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PSQLError.swift:443]:
```suggestion
    ///   - postgresData: The actual bytes on which decoding was tried
```

File: Sources/PostgresNIO/New/PSQLError.swift [hunk 4]
```diff
@@ -432,8 +432,18 @@ public struct PostgresDecodingError: Error, Equatable {
...
     public let line: Int
 
-    @usableFromInline
-    init(
+    /// Create a new ``PostgresDecodingError``
+    /// - Parameters:
+    ///   - code: The reason for the decoding error
+    ///   - columnName: The cell's column name for which the decoding failed
+    ///   - columnIndex: The cell's column index for which the decoding failed
+    ///   - targetType: The Swift type which the data should have been decoded to
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PSQLError.swift:440]:
```suggestion
    ///   - targetType: The Swift type to which the data was to be decoded
```

File: Sources/PostgresNIO/New/PSQLError.swift [hunk 5]
```diff
@@ -432,8 +432,18 @@ public struct PostgresDecodingError: Error, Equatable {
     /// The line the decoding was attempted in
     public let line: Int
 
-    @usableFromInline
-    init(
+    /// Create a new ``PostgresDecodingError``
+    /// - Parameters:
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PSQLError.swift:436]:
```suggestion

    /// - Parameters:
```

File: Sources/PostgresNIO/New/PSQLError.swift [hunk 6]
```diff
@@ -432,8 +432,18 @@ public struct PostgresDecodingError: Error, Equatable {
     /// The line the decoding was attempted in
     public let line: Int
 
-    @usableFromInline
-    init(
+    /// Create a new ``PostgresDecodingError``
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PSQLError.swift:435]:
```suggestion
    /// Create a new ``PostgresDecodingError``.
```

**Review by gwynne — CHANGES_REQUESTED** [review]
Bunch of mostly petty grammar nits 😛 (I do think the extra notes for `columnName` and `columnIndex` are worth adding, the rest is heavy pedantry.)

**Comment by fabianfett** [comment]
Not needed in PostgresKit.

**Pull request closed without merging**